src/main/fc/fc_rc.c

```diff
@@ -169,6 +169,8 @@ static void calculateSetpointRate(int axis)
 
         angleRate = applyRates(axis, rcCommandf, rcCommandfAbs);
     }
+    // Rate limit from profile (deg/sec)
+    setpointRate[axis] = constrainf(angleRate, -1.0f * currentControlRateProfile->rate_limit[axis], 1.0f * currentControlRateProfile->rate_limit[axis]);
```

Comment:
value is overwritten on next line. Either `angleRate = ` or beter make sure that `currentControlRateProfile->rate_limit[axis] <= SETPOINT_RATE_LIMIT` and remove next test

Reply:
Existing code uses `#define SETPOINT_RATE_LIMIT 1998.0f` in 'fc_rc.c' itself, I assume as a hard-coded limit to avoid triggering the 'yaw spin to the moon' issue. Although a 2deg/sec margin seems a bit close to me...

So I left that guard in place as-is so the `SETPOINT_RATE_LIMIT` can be updated independent of the new rate limits.

I prefer the independence of the constrains, instead of first comparing `currentControlRateProfile->rate_limit[axis]` to `SETPOINT_RATE_LIMIT` and then applying a single constrainf.

Is there a style guideline on this? I could not find it...

Reply:
Current code in this PR does not work - angleRate is used for SETPOINT_RATE_LIMIT,

The idea was to only apply `SETPOINT_RATE_LIMIT` as upper limit for `rate_limit[axis]` and assume that `rate_limit[axis]` is tight enough here ... but it's your call, both will work ...

Reply:
Agreed, I swapped the statements before my commit, as it looked cleaner. But it broke the functionality.
Anyway, I see that `SETPOINT_RATE_LIMIT` is also applied when in `FPV ANGLE MIX` mode, so this feature needs adding there too. I'll commit both changes for review.